typescript/src/api-doc-types/MethodMemberDoc.ts

```diff
@@ -1,28 +1,34 @@
 /* tslint:disable:no-bitwise */
 import { Declaration, SignatureDeclaration, Symbol, SymbolFlags } from 'typescript';
-import { getParameters } from '../services/TsParser/getParameters';
 import { getTypeParametersText } from '../services/TsParser/getTypeParametersText';
 import { encodeAnchor } from '../utils/encodeAnchor';
+import { CallableDoc, getParameters } from './CallableDoc';
 import { ContainerExportDoc } from './ContainerExportDoc';
 import { MemberDoc } from './MemberDoc';
+import { ModuleDoc } from './ModuleDoc';
+import { ParameterDoc } from './ParameterDoc';
 
-export class MethodMemberDoc extends MemberDoc {
-  readonly parameters = getParameters(this.declaration as SignatureDeclaration, this.namespacesToInclude);
+export class MethodMemberDoc extends MemberDoc implements CallableDoc {
   readonly name = this.computeName();
+  readonly parameterDocs = getParameters(this);
+  readonly parameters = this.parameterDocs.map(p => p.paramText);
   readonly anchor = this.computeAnchor();
-  readonly id = `${this.containerDoc.id}.${this.anchor})`;
+  readonly id = `${this.containerDoc.id}.${this.anchor}`;
   readonly aliases = this.computeAliases();
   readonly typeParameters = getTypeParametersText(this.declaration, this.namespacesToInclude);
 
   constructor(
     containerDoc: ContainerExportDoc,
     symbol: Symbol,
     declaration: Declaration,
-    basePath: string,
-    namespacesToInclude: string[],
     isStatic: boolean,
     public overloads: MethodMemberDoc[] = []) {
-    super(containerDoc, symbol, declaration, basePath, namespacesToInclude, isStatic);
+    super(containerDoc, symbol, declaration, isStatic);
+    // fix up parameter ids and aliases, now that we have computed the id for this doc
+    this.parameterDocs.forEach(param => {
+      param.id = `${param.callableDoc.id}~${param.name}`;
+      param.aliases = param.callableDoc.aliases.map(alias => `${alias}~${param.name}`);
+      });
```

Comment:
Wrong indentation here. Also it's kind of confusing, can't we use `this` instead of `param.callableDoc` here?

Reply:
Updated